log_request_id/middleware.py

```diff
@@ -21,18 +21,24 @@ def process_request(self, request):
         request.id = request_id
 
     def get_log_message(self, request, response):
+        message = 'method=%s path=%s status=%s' % (request.method, request.path, response.status_code)
+
+        if getattr(request, 'session', None) and request.session.is_empty():
+            # avoid accessing session if it is empty
+            return message
+
         user = getattr(request, 'user', None)
-        user_attribute = getattr(settings, LOG_USER_ATTRIBUTE_SETTING, False)
+        if not user:
+            return message
+
+        # `LOG_USER_ATTRIBUTE_SETTING` accepts False/None to skip setting attribute
+        #  but falls back to 'pk' if value is not set
+        user_attribute = getattr(settings, LOG_USER_ATTRIBUTE_SETTING, 'pk')
```

Comment:
I would move this bit, one step up. Check the setting before you do anything with the user.

Reply:
Fantastic proposal, Johannes 🙏